proposed/2023/sponsor-link.md

```diff
@@ -0,0 +1,164 @@
+
+Additionally, this information being surfaced can help encourage developers who see these requests to advocate for corporate sponsorship budget of critical packages they use in their line of work. Today there are two major sources of funding: software companies and other developers. This proposal hopes to bring more awareness to both scenarios but focuses mostly on the latter.
+
+There is a need to support the free public infrastructure. But right now there is no revenue stream around the time being put into it. People scream if their clocks are off by a second or a security vulnerability hasn't been addressed in a critical OSS package. They say, "Yes, we need you, but we can't give you any money". 
+
+Maintainers don't work on OSS for money. They don't do it for fame. They do it out of pride in craftsmanship and the responsibility for something they believe in. It takes nerves of steel to work for many years on hundreds of thousands of lines of very complex code, with every line of code you touch being visible to the world, knowing that code is used by banks, weapon systems, smartphones, industry, government, and just about everywhere. With the knowing that you'll be ignored and unappreciated until something goes wrong. 
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+#### .nuspec / .csproj MSBuild property
+
+The `.nuspec` will introduce a new collection element called `<sponsors>` which is a collection of zero or more `<sponsor>` elements specifying the sponsor requirements for the package.
+
+```xml
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+    <metadata minClientVersion="100.0.0.1">
+        <id>dasdas</id>
+        <version>2.0.0</version>
+        <title />
+        <authors>dsadas</authors>
+        <owners />
+        <requireLicenseAcceptance>false</requireLicenseAcceptance>
+        <description>My package description.</description>
+    </metadata>
+    <sponsors>
+        <sponsor type="Patreon" url="https://www.patreon.com/mypatreonusername" />
+        <sponsor type="GitHub Sponsors" url="https://github.com/sponsors/mygithubusername" />
+        <sponsor type="Tidelift" url="https://tidelift.com/subscription/pkg/mypackagename" />
+    </sponsors>
+</package>
+```
+
+For .NET projects that use the `PackageReference` format which draws inputs from the project file, one can create a new `<ItemGroup>` for their sponsors such as:
+
+```xml
+<ItemGroup>
+    <Sponsors Include="Patreon" Url="https://www.patreon.com/mypatreonusername"/>
+    <Sponsors Include="GitHub Sponsors" Url="https://github.com/sponsors/mygithubusername"/>
+    <Sponsors Include="Tidelift" Url="https://tidelift.com/subscription/pkg/mypackagename"/>
```

Comment:
Can we support a `Custom` type as well? For example, some people prefer to sponsor projects through the means of crypto donations, so it would be nice to include a personal webpage that contains a list of your wallets.

Reply:
I think having a fallback plain link type would still be nice regardless, at least to support lesser known donation platforms

Reply:
For the support of custom sponsorship types, it's a two part question. One is what is supported in the data model (i.e. what will NuGet pack allow in the .nuspec or .csproj metadata) and what is supported on package sources (e.g. NuGet.org). I think we should be permissive in the data model (e.g. freeform or even optional type) and allow package sources to be a bit stricter (e.g. NuGet.org may say it must be an HTTPS URL and have a set of allowed types or no type at all giving a well understood fallback experience).

This is the approach we have taken with package type and repository metadata and it's served us okay. There are drawbacks of a looser data model but I think it's best to encourage innovation in the values rather than lock down to a fixed set that we have to revise every so often and with dubious criteria.

Reply:
> I was mostly thinking of reputable sponsorship platforms which may or may not support it.

Not all countries have access to what you most likely call a "reputable sponsorship platform" (such as patreon and github sponsors).

Reply:
When I see URLs in package metadata, I immediately think about how spammers would abuse it. Not sure if it is in scope of the spec, but we should consider using `rel='nofollow'` for platforms unknown to Gallery if we choose to render them in the first place.